### PR TITLE
Add replica set example demo project

### PR DESCRIPTION
**Pull request by drubin**
This is an example application to show off how to use mongo with various hydra configurations. 

For now, this configuration doesn't actually use mongo-hydra instead it uses a place holder docker-image and a manual step to setup the replica set. 

This is in draft while we're working making this possible with hydra.

### CODE REVIEW

File: examples/replicaset/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+version: '3.8'
+services:
+  rs-0:
```
Comment (pelzerim) [inline examples/replicaset/docker-compose.yaml:3]: Can we use a better naming maybe? People will just copy paste this and then forever be stuck with rs-0 :P  `mongod-replica-0`? Same for the --replSet

File: examples/replicaset/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Replicaset example
+
+Create base replica configuration and then set it up
```
Comment (pelzerim) [inline examples/replicaset/README.md:3]:
```suggestion
This example shows how to create a simple MongoDB replica set with docker-compose and how to use mongo-hydra to set it up.
```

File: examples/replicaset/docker-compose.yaml (hunk 1 above)
Comment (drubin) [reply examples/replicaset/docker-compose.yaml:3]: How is that better than what I did? I am happy to rename the example rs to something else but not sure replica is even better... 

Why prefix it with `mongod` its obviously a mongo replicaset you are using mongo to connect to it.

Comment (pelzerim) [reply examples/replicaset/docker-compose.yaml:3]: I imagine these running in a larger environment (lets say k8s) and then you have a bunch of pods and it is very helpful to prefix them by something, so they are alphabetically ordered. Doesn't super much matter by what its prefixed.

How about this naming convention:

`mongo-${rs || config || shard-a}-${replica number}` 

replica sets get rs
config servers config
shards `shard-` letters

Comment (drubin) [reply examples/replicaset/docker-compose.yaml:3]: This I am by the mongod was the issue didn't really see the need

**Comment by drubin** [comment]
Do we still want to have a nodejs example? I feel like even if  we do want this example we should start from scratch and make dedicated client examples. i.e mongoose/js /python / go or something.

**Comment by drubin** [comment]
Closing this we  can  revisit it again later this is impossible to  rebase and things will change.

**Pull request closed without merging**